website/docs/layouts/flex/partials/code/how-to-use.md

```diff
@@ -0,0 +1,573 @@
+  <Doc::placeholder @width="300px" @height="40px" @background="#fff8d2">Extra content</Doc::placeholder>
+  <Doc::placeholder @width="240px" @height="40px" @background="#f3d9c5">Even more content</Doc::placeholder>
+</Hds::Layout::Flex>
+```
+In this case:
+
+- the `@wrap=\{{true}}` applied to the flex container allows the flex items to wrap on multiple lines
+- the `@gap=\{{array "8" "24"}}` applied to the flex container defines the gap between the items along the two "directions" (the first value is the space between "rows", the second between "columns")
+
+### Nesting
+
+It's possible to nest flexbox containers to achieve more complex layouts:
+
+```handlebars
+<Hds::Layout::Flex @direction="column" class="doc-flex-outlined-container">
+  <Doc::placeholder @height="50px" @background="#e5ffd2">Top content</Doc::placeholder>
+  <Hds::Layout::Flex @direction="row" as |LF|>
+    <Doc::placeholder @width="150px" @height="auto" @background="#e4c5f3">Side content</Doc::placeholder>
+    <LF.Item @grow={{true}}>
+      <Doc::placeholder @width="auto" @height="250px" @background="#d2f4ff">Main content</Doc::placeholder>
+    </LF.Item>
+  </Hds::Layout::Flex>
+  <Doc::placeholder @height="50px" @background="#fff8d2">Bottom content</Doc::placeholder>
+</Hds::Layout::Flex>
+```
+
+In this case:
+
+- the outer container is a column-based flexbox layout, with three child items
+- the second item is a row-based flexbox layout, used to create two side-by side content blocks; of these two the first has a fixed width, while the second fills the remaining area thanks to the `@grow=\{{true}}` attribute
+
+!!! Insight
+
+Depending on the complexity of the design you need to implement, you may want to consider using a CSS `grid` layout, instead of a `flex` layout.
+
+!!!
+
+---
+
+## Common layout patterns
+
+Below some more realistic examples of common layouts that can be achieved using the `Layout::Flex` component, in composition with other HDS components.
+
+!!! Warning
+
+**Important**
+
+The examples below are meant to show how one _could_ use the `Layout::Flex` component to implement certain common/standard UI patterns. They're **not** meant to be taken literally as they are and be used in production code. Also, some of these patterns are already implemented directly in HDS components that are ready to use.
+
+!!!
+
+### Media + Text/Content
+
+This is a classic layout, where a "media" element (it could be an image, an icon, an avatar, etc) is laid out next to a block of text (but could be more complex content). The two elements are visually centered along their horizontal axis.
+
+
+```handlebars
+<Hds::Layout::Flex @align="center" @gap="8">
+  <img class="doc-flex-media-avatar" src="/assets/images/avatar.png" alt="portrait of a cat wearing coat and tie" />
+  <Hds::Text::Body @size="200" @tag="p">Lorem ipsum dolor sit amet</Hds::Text::Body>
+</Hds::Layout::Flex>
+
+<br/>
```

Comment:
Super nit: it bothers me to see `<br/>` tags used as spacers like this... I'd personally avoid including those in code examples that others may copy.

Reply:
the alternative is to have `doc-something` classes just for this, which is also not great, no? if you have strong feelings I can change it though.

Reply:
It just really bugs me to model something in an example that I wouldn't recommend people use but it's probably just me

Reply:
Fixed in https://github.com/hashicorp/design-system/pull/2764/commits/e89a70e4460f05b13b27a5d87616195d33b1173b